chunk-test.py

```diff
@@ -0,0 +1,91 @@
+import os
+import random
+import shutil
+import tempfile
+from typing import BinaryIO, Dict, Iterable
+import uuid
+
+# Input data from the client
+
+chunks = [
+    dict(fileName="fileA", chunkNumber=1, chunkContents=b"chunkA1\n",),
+    dict(fileName="fileA", chunkNumber=2, chunkContents=b"chunkA2\n",),
+    dict(fileName="fileB", chunkNumber=1, chunkContents=b"chunkB1\n",),
+    dict(fileName="fileB", chunkNumber=2, chunkContents=b"chunkB2\n",),
+    dict(fileName="fileB", chunkNumber=3, chunkContents=b"chunkB3\n",),
+]
+random.shuffle(chunks)
+
+chunked_upload_id = str(uuid.uuid4())
+
+
+# Overall approach:
+# For each file we want to upload:
+# 1. Upload each chunk to a separate file
+# 2. When we're done uploading all the chunks, concatenate the chunks into a single file
+# 3. Open the concatenated file as a file stream
```

Comment:
Combine these so we don't have to duplicate the file data?

Reply:
I tried this at first but it got a little weird. There's no built in way to do it and so you have to get kind of fancy/hacky to meet the BinaryIO interface that the consuming code expects. I think it could be a cool follow up once we get it working